devops/ci/multi_node_deploy/3node_deploy.jenkinsfile

```diff
@@ -128,31 +127,28 @@ node {
                     \"\"\" | sed -e 's/^[ \t]*//' > /root/config.ini
                 """
                 for (remote in remotes) {
-                    device_list = sshCommand remote: remote, command: "lsblk -nd -o NAME -e 11|grep -v sda|sed 's|sd|/dev/sd|g'|paste -s -d, -"
                     if (remote.name.equals(remotes[0].name) ) {
                         sshCommand remote: remotes[0], command: """
+                            device_list=\$(lsblk -nd -o NAME -e 11|grep -v sda|sed 's|sd|/dev/sd|g'|paste -s -d, -)
```

Comment:
This is generic and the query has to happen on all nodes, not just primary.
We are capturing device list from each node and then populating it in config.ini.
This change would only fill details from primary node.
To fix that issue, you have duplicated the command in else block :)